src/apis/auth.ts

```diff
@@ -0,0 +1,38 @@
+import axios from 'axios';
+
+import { TOKEN_KEY } from '../constants/auth';
+import { setStorage } from '../utils/storage';
+import http from './instance';
+
+export const signin = async ({
+  email,
+  password,
+}: {
+  email: string;
+  password: string;
+}) => {
+  let isSignInFailed = false;
+  let errorMessage = '';
+
+  try {
+    const {
+      data: { token },
+    } = await http.post({
+      url: '/login',
+      data: {
+        email,
+        password,
+      },
+    });
+
+    token && setStorage(TOKEN_KEY, token);
+  } catch (error) {
+    if (error && axios.isAxiosError(error)) {
+      isSignInFailed = true;
+      errorMessage = error.response?.data;
+    }
+    console.error(error);
+  }
+
+  return { isSignInFailed, errorMessage };
+};
```

Comment:
```suggestion
  try {
    const {
      data: { token },
    } = await http.post({
      url: '/login',
      data: {
        email,
        password,
      },
    });
    token && setStorage(TOKEN_KEY, token);
  } catch (error) {
    if (error && axios.isAxiosError(error))
       return { isSignInFailed : true, errorMessage : error.response?.data }
  }
  return { isSignInFailed : false , errorMessage : '' };
};
```

대충 이런식으로 혹시 let을 지워보는 건 어떤가요?

Reply:
더 간결해질 것 같아 위처럼 변경했어요👍 console.error(error)는 나중에 디버깅을 위해서 놔두었습니다.